### PR TITLE
Cut Website Documentation for RNW v0.73

### PR DESCRIPTION
**Pull request by chiaramooney**
### Why
Update website for RNW v0.73. 
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows-samples/pull/903)

### CODE REVIEW

File: samples/NativeModuleSample/yarn.lock
Comment (jonthysell) [inline samples/NativeModuleSample/yarn.lock:1]: This shouldn't be here.

File: samples/NativeModuleSample/cppwinrt/codegen/NativeAsyncMethodExamplesSpec.g.h [hunk 1]
```diff
@@ -13,9 +13,17 @@
 
 namespace NativeModuleSample {
 
+REACT_STRUCT(AsyncMethodExamplesSpec_Response)
```
Comment (jonthysell) [inline samples/NativeModuleSample/cppwinrt/codegen/NativeAsyncMethodExamplesSpec.g.h:16]: While these changes may be necessary to align with the docs when the sample is updated to 0.73, it shouldn't really be a part of the docs cut.

File: website/pages/en/support.js [hunk 2]
```diff
@@ -16,8 +16,8 @@ The React Native for Windows (RNW) Team strives to provide full support for the
 | Version | Support Phase | Release Date | Active Support Start | Maintenance Support Start | End of Support |
 | -- | -- | -- | -- | -- | -- |
 | [main](https://www.npmjs.com/package/react-native-windows/v/canary) | [Canary](#canary-support) | *N/A* | *N/A* | *N/A* | *N/A* |
-| [0.73](https://www.npmjs.com/package/react-native-windows/v/preview) | [Preview](#preview-support) | *TBD* | *TBD* | *TBD* | *TBD* |
-| [0.72](https://www.npmjs.com/package/react-native-windows/v/latest) | [Active](#active-support) | 06/23/2023 | 06/23/2023 | *TBD* | *TBD* |
+| [0.73](https://www.npmjs.com/package/react-native-windows/v/latest) | [Active](#active-support) | 12/11/23 | 12/11/23 | *TBD* | *TBD* |
+| [0.72](https://www.npmjs.com/package/react-native-windows/v/v0.72-stable) | [Active](#active-support) | 06/23/2023 | 06/23/2023 | 01/11/23 | 03/11/23 |
```
Comment (jonthysell) [inline website/pages/en/support.js:20]: should be the last day of the next month, also the correct year
```suggestion
| [0.72](https://www.npmjs.com/package/react-native-windows/v/v0.72-stable) | [Active](#active-support) | 06/23/2023 | 06/23/2023 | 01/31/24 | 03/30/24 |
```

**Review by jonthysell — APPROVED** [review]

**Comment by TatianaKapos** [comment]
The PR: https://github.com/microsoft/react-native-windows/pull/12371 makes Hermes the default engine in 0.73, we will want to fix the hermes.md file before cutting the documentation

**Review by TatianaKapos — APPROVED** [review]

**Pull request merged**